src/main/java/net/rptools/maptool/client/functions/TokenImage.java

```diff
@@ -171,6 +173,34 @@ public Object childEvaluate(
       }
     }
 
+    StringBuilder assetId = new StringBuilder("asset://");
+    if (functionName.equalsIgnoreCase("createAsset")) {
+      FunctionUtil.checkNumberParam(functionName, args, 2, 2);
+      String imageName = args.get(0).toString();
+      String imageString = args.get(1).toString();
+      if (imageName == "" || imageString == "") {
+        throw new ParserException(
+            I18N.getText("macro.function.general.paramCannotBeEmpty", functionName));
+      } else {
+        if (imageString.length() > 8) {
+          byte[] imageBytes = Base64.decode(imageString);
+          String imageCheck = new String(imageBytes, 0, 4);
+          /* header check for: webp || jpg || png */
+          if (imageCheck.equals("RIFF") || imageCheck.equals("ÿØÿà") || imageCheck.equals("‰PNG")) {
```

Comment:
If you make these constants so its more meaningful when people read the code I will approve it :)

Reply:
Is this acceptable or _inline_ with your thoughts?
```
public static final String FILE_HEADER_WEBP = "RIFF";
public static final String FILE_HEADER_JPG = "ÿØÿà";
public static final String FILE_HEADER_PNG = "‰PNG";
  ....
if (imageCheck.equals(FILE_HEADER_WEBP) || imageCheck.equals(FILE_HEADER_JPG) || imageCheck.equals(FILE_HEADER_PNG)) {```

Reply:
Is this in-line with what you're wanting or am I _off_ :) ?
``` if (imageCheck.equals(FILE_HEADER_WEBP) || imageCheck.equals(FILE_HEADER_JPG) || imageCheck.equals(FILE_HEADER_PNG)) {```

Reply:
spotless, I hate thee . . .